src/App/Pages/ProductDetail/ProductDetail.tsx

```diff
@@ -1,103 +1,87 @@
-import { useEffect, useState } from "react";
+import { useEffect } from "react";
 
 import Button from "@components/Button";
 import { ButtonColor } from "@components/Button/Button";
-import ProductList from "@components/ProductList/ProductList";
-import axios from "axios";
-import { useParams } from "react-router-dom";
+import Card from "@components/Card";
+import Loader from "@components/Loader";
+import { Meta } from "@utils/meta";
+import { observer } from "mobx-react-lite";
+import { useNavigate, useParams } from "react-router-dom";
 
-import ProductType from "../../../customTypes/productType";
+import { useRootStore } from "../../../context/StoreContext";
 import styles from "./ProductDetail.module.scss";
 
 
 const ProductDetail = () => {
   // Получаем из url id товара
   // (id, поскольку записали :id в path роута)
-  const [product, setProduct] = useState<ProductType>();
   const { id } = useParams();
 
-  useEffect(() => {
-    fetch();
-  }, [id]);
+  const { productDetailStore } = useRootStore();
 
-  const fetch = async () => {
-    const result = await axios({
-      method: "get",
-      url: "https://fakestoreapi.com/products/" + id,
-    });
-    setProduct({
-      id: result.data.id,
-      title: result.data.title,
-      category: result.data.category,
-      description: result.data.description,
-      image: result.data.image,
-      price: result.data.price,
-      rating: result.data.rating,
-    });
-  };
+  useEffect(() => {
+    if (id) {
+    productDetailStore.getProductDetailByID(id.toString());}
+  }, [id])
 
-  //стоит заменить работу с undefined на null, ибо undefined - нечто непреднамеренное,
-  // а null сразу даёт понять, что мы предполагаем, что product может быть ничем
 
+  let navigate = useNavigate();
 
-  //заглушка чтобы не ругалось
-  function getProduct() {
-    if (product !== undefined) {
-      return (
+  return (
+    <div>
+      {(productDetailStore.meta === Meta.error) && <div>Такого нет уходите</div>}
+      {(productDetailStore.meta === Meta.loading) ? <Loader className={styles.product__loader}/> :
+      (productDetailStore.product && productDetailStore.meta === Meta.success) &&
         <div className={styles.product__display}>
           <img
-            src={product.image}
+            src={productDetailStore.product.image}
             alt={"изображение отсутствует"}
```

Comment:
во-первых:
```suggestion
            alt="изображение отсутствует"
```

во-вторых, в качестве alt стоит указывать не некий fallback-вариант с объяснением ошибки, а описание картинки, как если бы она была. это полезно, например, для разработчиков, чтобы они могли увидеть, что означает это изображение. этот текст может появиться, если изображение не подгрузится или ещё какая ошибка произойдёт. да, можно бы было отобразить ошибку с помощью этого текста, но этот текст также видят люди с нарушением зрения, использующие скринридеры